### PR TITLE
Mo commenting

### PR DESCRIPTION
**Pull request by jaseychanders**

### CODE REVIEW

File: src/main/java/org/uacr/models/state/ParallelState.java [hunk 1]
```diff
@@ -23,26 +26,38 @@
     private final Set<State> fBackgroundStates;
     private final String fStateName;
 
+    /**
+     * @param modelFactory so the parallel state can creat its substates
+     * @param name of the parallel state
+     * @param parser ymal parser for the parallel state
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/models/state/ParallelState.java:32]: yaml misspelled

File: src/main/java/org/uacr/models/state/SingleState.java [hunk 2]
```diff
@@ -14,9 +14,9 @@
 
 /**
  * The base for all states
+ * States preform a single action on a single subsystem
  * One implementation is created for each state listed under 'single_state' in the state ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/models/state/SingleState.java:18]: yaml misspelled

File: src/main/java/org/uacr/models/state/TimedState.java [hunk 3]
```diff
@@ -25,6 +27,13 @@
     private final String fStateName;
     private final String fSubStateName;
 
+    /**
+     * @param modelFactory so it can create its substate
+     * @param name of the TimedState
+     * @param parser ymal parser for the TimedState
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/models/state/TimedState.java:33]: yaml misspelled

File: src/main/java/org/uacr/robot/AbstractModelFactory.java [hunk 4]
```diff
@@ -43,6 +50,16 @@ public AbstractModelFactory(InputValues inputValues, OutputValues outputValues,
         fModelFactories = new ArrayList<>();
     }
 
+    /**
+     * Creates an Output Numeric (an output that takes a numeric value such as a motor)
+     * Loops through all implementations of AbstractModelFactory and tries to create the desired object with each until it finds the correct implementation
+     * This is to allow for sim and robot model factories
+     * @param name of the object to create
+     * @param config for the object, usually contains things like ID, weather it is inverted etc.
+     * @param parser the ymal parser for the object, used to read the config
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/robot/AbstractModelFactory.java:59]: yaml misspelled

File: src/main/java/org/uacr/robot/AbstractModelFactory.java [hunk 5]
```diff
@@ -53,6 +70,16 @@ public OutputNumeric createOutputNumeric(Object name, Config config, YamlConfigP
         throw new ConfigurationTypeDoesNotExistException(config.getType());
     }
 
+    /**
+     * Creates an Output Boolean (an output that takes a boolean value such as a solenoid)
+     * Loops through all implementations of AbstractModelFactory and tries to create the desired object with each until it finds the correct implementation
+     * This is to allow for sim and robot model factories
+     * @param name of the object to create
+     * @param config for the object, usually contains things like ID, weather it is inverted etc.
+     * @param parser the ymal parser for the object, used to read the config
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/robot/AbstractModelFactory.java:79]: yaml misspelled

File: src/main/java/org/uacr/robot/AbstractModelFactory.java [hunk 6]
```diff
@@ -103,6 +166,14 @@ public Behavior createBehavior(String name, Config config) {
         throw new ConfigurationTypeDoesNotExistException(config.getType());
     }
 
+    /**
+     * Creates a State (an instance of a Behavior that preforms a specific task)
+     * @param name of the object to create
+     * @param parser the ymal config parser used to read the config values
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/robot/AbstractModelFactory.java:172]: yaml misspelled

File: src/main/java/org/uacr/services/input/InputService.java [hunk 7]
```diff
@@ -56,6 +63,14 @@ public InputService(AbstractModelFactory modelFactory, InputValues inputValues,
         mFrameCycleTimeThreshold = -1;
     }
 
+    /**
+     * Starts the input service
+     * Obtains a list of all input objects (boolean, numeric and vector)
+     * Loads the input ymal files
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/services/input/InputService.java:69]: yaml misspelled

File: src/main/java/org/uacr/services/output/OutputService.java [hunk 8]
```diff
@@ -52,6 +60,15 @@ public OutputService(AbstractModelFactory modelFactory, FMS fms, InputValues inp
         mFrameTimeThreshold = -1;
     }
 
+
+    /**
+     * Starts the OutputService
+     * Obtains a list of all output objects (boolean and numeric)
+     * Loads the output ymal files
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/services/output/OutputService.java:67]: yaml misspelled

File: src/main/java/org/uacr/shared/abstractions/RobotConfiguration.java [hunk 9]
```diff
@@ -4,6 +4,10 @@
 import java.util.Map;
 import java.util.Set;
 
+/**
+ * Stores all the configuration values from the robot-configuration.ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/abstractions/RobotConfiguration.java:8]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 10]
```diff
@@ -60,6 +68,13 @@ public SharedObjectsDirectory(AbstractModelFactory modelFactory, RobotConfigurat
 
     //--------------------------- Inputs ----------------------------------------//
 
+    /**
+     * Loops through all inputs and calls the appropriate method to create them and store them in the appropriate map
+     * Once all inputs have been created, calls initialize on all inputs
+     * @param inputBooleansParser holds the information from the InputBooleans ymal file
+     * @param inputNumericsParser holds the information from the InputNumerics ymal file
+     * @param inputVectorsParser holds the information from the InputVectors ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:76]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 11]
```diff
@@ -97,37 +112,76 @@ public void registerAllInputs(YamlConfigParser inputBooleansParser, YamlConfigPa
 
     }
 
+    /**
+     * Uses the ModelFactory to create the desired InputBoolean and stores it in the InputBooleanObjects map
+     * @param name of the input to be created
+     * @param config the ymal configuration for the input
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:118]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 12]
```diff
@@ -97,37 +112,76 @@ public void registerAllInputs(YamlConfigParser inputBooleansParser, YamlConfigPa
...
+
     @Override
     public void registerInputBoolean(String name, Config config) {
         fInputBooleanObjects.put(name, fModelFactory.createInputBoolean(name, config));
     }
 
+    /**
+     * Uses the ModelFactory to create the desired InputNumeric and stores it in the InputBooleanObjects map
+     * @param name of the input to be created
+     * @param config the ymal configuration for the input
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:129]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 13]
```diff
@@ -97,37 +112,76 @@ public void registerAllInputs(YamlConfigParser inputBooleansParser, YamlConfigPa
...
+
     @Override
     public void registerInputNumeric(String name, Config config) {
         fInputNumericObjects.put(name, fModelFactory.createInputNumeric(name, config));
     }
 
+    /**
+     * Uses the ModelFactory to create the desired InputVector and stores it in the InputBooleanObjects map
+     * @param name of the input to be created
+     * @param config the ymal configuration for the input
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:140]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 14]
```diff
@@ -97,37 +112,76 @@ public void registerAllInputs(YamlConfigParser inputBooleansParser, YamlConfigPa
...
     @Override
     public InputVector getInputVectorObject(String name) {
         return fInputVectorObjects.get(name);
     }
 
     //--------------------------- States ----------------------------------------//
+
+    /**
+     * Loops through all states and calls the method to create them and store them
+     * @param statesParser holds the information from the States ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:182]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 15]
```diff
@@ -136,22 +190,46 @@ public void registerAllStates(YamlConfigParser statesParser) {
         }
     }
 
+    /**
+     * Uses the ModelFactory to create the desired state
+     * @param name of the state to be created
+     * @param statesParser holds the information from the States ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:196]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 16]
```diff
@@ -136,22 +190,46 @@ public void registerAllStates(YamlConfigParser statesParser) {
...
     public void setStateObject(String name, State state) {
         fStateObjects.put(name, state);
     }
 
     //--------------------------- Outputs ----------------------------------------//
+
+    /**
+     * Loops through all outputs and calls the appropriate method to create them and store them in the appropriate map
+     * @param outputNumericsParser holds the information from the OutputNumerics ymal file
+     * @param outputsBooleanParser holds the information from the OutputBooleans ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:230]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 17]
```diff
@@ -167,33 +245,65 @@ public void registerAllOutputs(YamlConfigParser outputNumericsParser, YamlConfig
         }
     }
 
+    /**
+     * Uses the ModelFactory to create the desired OutputNumeric and stores it in the OutputNumericObjects map
+     * @param name of the output to be created
+     * @param config the ymal configuration for the output
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:251]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java [hunk 18]
```diff
@@ -167,33 +245,65 @@ public void registerAllOutputs(YamlConfigParser outputNumericsParser, YamlConfig
...
+
     @Override
     public void registerOutputNumeric(String name, Config config, YamlConfigParser parser) {
         fOutputNumericObjects.put(name, fModelFactory.createOutputNumeric(name, config, parser));
     }
 
+    /**
+     * Uses the ModelFactory to create the desired OutputBoolean and stores it in the OutputBooleanObjects map
+     * @param name of the output to be created
+     * @param config the ymal configuration for the output
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedObjectsDirectory.java:262]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 19]
```diff
@@ -11,6 +11,10 @@
 import javax.annotation.Nullable;
 import java.util.*;
 
+/**
+ * Stores and allows access to all information in the robot-configuration.ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:15]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 20]
```diff
@@ -76,9 +96,14 @@ public void initialize() {
             }
         }
 
+        // A map containing all states grouped by priority level
         return stateMap;
     }
 
+    /**
+     * @return a set of all the state names listed in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:104]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 21]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
         return stateSet;
     }
 
+    /**
+     * @return the list of subsystem names specified in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:144]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 22]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
...
     @Override
     public Set<String> getSubsystemNames() {
         return getSet("general", "subsystems");
     }
 
+    /**
+     * @return the list of InputBoolean names specified in robot-configuration.ymal
+     */
+
     @Override
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:155]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 23]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
...
     public Set<String> getInputBooleanNames() {
         return getSet("general", "input_booleans");
     }
 
+    /**
+     * @return the list of InputNumeric names specified in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:161]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 24]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
...
+     */
+
     @Override
     public Set<String> getInputNumericNames() {
         return getSet("general", "input_numerics");
     }
 
+    /**
+     * @return the list of InputVector names specified in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:170]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 25]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
...
+     */
+
     @Override
     public Set<String> getInputVectorNames() {
         return getSet("general", "input_vectors");
     }
 
+    /**
+     * @return the list of OutputNumeric names specified in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:179]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 26]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
...
+     */
+
     @Override
     public Set<String> getOutputNumericNames() {
         return getSet("general", "output_numerics");
     }
 
+    /**
+     * @return the list of OutputBoolean names specified in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:188]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 27]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
...
+     */
+
     @Override
     public Set<String> getOutputBooleanNames() {
         return getSet("general", "output_booleans");
     }
 
+    /**
+     * Retrieves the value of a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:197]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 28]
```diff
@@ -115,48 +140,88 @@ public void initialize() {
...
+     * @return a map of all the key value pairs in the specified category
+     */
     @Override
     public Map<String, Object> getCategory(String category) {
         ensureCategoryExists(category);
         return mData.get(category);
     }
 
+    /**
+     * Retrieves an int from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:219]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 29]
```diff
@@ -171,6 +236,13 @@ public int getInt(String category, String key) {
         }
     }
 
+    /**
+     * Retrieves a double from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:240]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 30]
```diff
@@ -185,6 +257,13 @@ public double getDouble(String category, String key) {
         }
     }
 
+    /**
+     * Retrieves a boolean from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:261]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 31]
```diff
@@ -199,6 +278,13 @@ public boolean getBoolean(String category, String key) {
         }
     }
 
+    /**
+     * Retrieves a String from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:282]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 32]
```diff
@@ -213,6 +299,13 @@ public String getString(String category, String key) {
         }
     }
 
+    /**
+     * Retrieves a List from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:303]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 33]
```diff
@@ -223,6 +316,13 @@ public String getString(String category, String key) {
         }
     }
 
+    /**
+     * Retrieves a Map from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:320]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 34]
```diff
@@ -233,6 +333,13 @@ public String getString(String category, String key) {
         }
     }
 
+    /**
+     * Retrieves a Set from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:337]: yaml misspelled

File: src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java [hunk 35]
```diff
@@ -243,6 +350,13 @@ public String getString(String category, String key) {
         }
     }
 
+
+    /**
+     * Retrieves an Enum from a key value pair specified under a category in robot-configuration.ymal
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/shared/concretions/SharedRobotConfiguration.java:355]: yaml misspelled

File: src/main/java/org/uacr/models/state/SequencerState.java [hunk 36]
```diff
@@ -27,34 +28,55 @@
     private State mCurrentState;
     private int mCurrentStateIndex;
 
+    /**
+     * @param modelFactory so it can create its substates
+     * @param name of the sequencer state
+     * @param parser the ymal parser for the sequencer
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/models/state/SequencerState.java:34]: yaml misspelled

File: src/main/java/org/uacr/models/behavior/Behavior.java [hunk 37]
```diff
@@ -15,7 +20,6 @@
      * Initializes the behavior to run the requested state
      * This is a place to read values from the state config file
      * This is called by the StateMachine on the first frame a state becomes active
-     *
      * @param stateName The name of the state being initialized
      * @param config    Contains the information under the name of the state in the state.ymal file
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/models/behavior/Behavior.java:24]: yaml misspelled

File: src/main/java/org/uacr/models/state/DoneForTimeState.java [hunk 38]
```diff
@@ -26,6 +33,13 @@
     private final String fStateName;
     private final String fSubStateName;
 
+    /**
+     * @param modelFactory so the doneForTimeState can create its substate
+     * @param name the name of the DoneForTimeState
+     * @param parser the ymal parser for the DoneForTimeState
```
Comment (HarrisonKaufman) [inline src/main/java/org/uacr/models/state/DoneForTimeState.java:39]: yaml misspelled

File: src/main/java/org/uacr/models/state/TimedState.java [hunk 39]
```diff
@@ -13,6 +13,8 @@
 /**
  * A shell that runs a state until it is done or it times out
  * I'm not sure this is useful in the current version of the framework
```
Comment (coychanders) [inline src/main/java/org/uacr/models/state/TimedState.java:15]: Is this still true?

File: src/main/java/org/uacr/models/state/TimedState.java [hunk 40]
```diff
@@ -13,6 +13,8 @@
...
+ * Accepts the substate and a single parameter the timout
+ * The time out is how long you want to state to run for at max
```
Comment (coychanders) [inline src/main/java/org/uacr/models/state/TimedState.java:17]: Could use a cleared explination

File: src/main/java/org/uacr/robot/AbstractModelFactory.java [hunk 41]
```diff
@@ -43,6 +50,16 @@ public AbstractModelFactory(InputValues inputValues, OutputValues outputValues,
         fModelFactories = new ArrayList<>();
     }
 
+    /**
+     * Creates an Output Numeric (an output that takes a numeric value such as a motor)
+     * Loops through all implementations of AbstractModelFactory and tries to create the desired object with each until it finds the correct implementation
+     * This is to allow for sim and robot model factories
+     * @param name of the object to create
+     * @param config for the object, usually contains things like ID, weather it is inverted etc.
```
Comment (coychanders) [inline src/main/java/org/uacr/robot/AbstractModelFactory.java:58]: whether
do a search as the same misspelling exists in other places

File: src/main/java/org/uacr/services/output/OutputService.java [hunk 42]
```diff
@@ -67,6 +84,14 @@ public void startUp() throws Exception {
         sLogger.trace("OutputService started");
     }
 
+    /**
+     * Runs every frame
+     * Loops through all outputs and updates output values using he values in the sharedOutputValues map
```
Comment (coychanders) [inline src/main/java/org/uacr/services/output/OutputService.java:89]: the instead of he

File: src/main/java/org/uacr/robot/AbstractModule.java [hunk 43]
```diff
@@ -3,6 +3,12 @@
 import org.uacr.shared.abstractions.*;
 import org.uacr.shared.concretions.*;
 
+/**
+ * Used by the injector decided witch concretion to use for each requested abstraction
```
Comment (coychanders) [inline src/main/java/org/uacr/robot/AbstractModule.java:7]: Use by the injector **to decide which**

**Comment by jaseychanders** [comment]
This is as far as I'm going to make it on commenting. We should pull it in as soon and it has been reviewed

File: src/main/java/org/uacr/models/state/DoneForTimeState.java [hunk 44]
```diff
@@ -54,17 +76,30 @@ public void initialize() {
         }
     }
 
+    /**
+     * Called every frame
+     */a
```
Comment (coychanders) [inline src/main/java/org/uacr/models/state/DoneForTimeState.java:81]: ? */a

File: src/main/java/org/uacr/robot/AbstractModule.java [hunk 45]
```diff
@@ -3,6 +3,12 @@
 import org.uacr.shared.abstractions.*;
 import org.uacr.shared.concretions.*;
 
+/**
+ * Used by the injector decided which concretion to use for each requested abstraction
```
Comment (coychanders) [inline src/main/java/org/uacr/robot/AbstractModule.java:7]: to decide

**Pull request merged**